### PR TITLE
Adding autocfg package

### PR DESCRIPTION
**Pull request by giswqs**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

Relevant issues:

- https://github.com/autogluon/autogluon/issues/612
- https://github.com/zhreshold/autocfg/issues/1

@arturdaraujo

### CODE REVIEW

**Comment by giswqs** [comment]
@conda-forge/help-python The recipe has passed all checks and is ready for review. Thanks.

File: recipes/autocfg/meta.yaml
Comment (dhirschfeld) [inline recipes/autocfg/meta.yaml:38]: The copy of this file can be removed here.

By referencing the file name/location under `license_file`, conda will automatically include a copy from the source at build time.

**Comment by giswqs** [comment]
@dhirschfeld The source file on PyPI does not include the LICENSE, so I think it needs to be included here. 

https://pypi.org/project/autocfg/#files
image

**Comment by giswqs** [comment]
@conda-forge-admin, please ping team

**Comment by giswqs** [comment]
@ocefpaf It is ready to be merged.

**Comment by dhirschfeld** [comment]
> The source file on PyPI does not include the LICENSE, so I think it needs to be included here.

If the sdist doesn't include the license then you do need to include a copy here however that should probably be fixed upstream (though it doesn't need to hold up merging this PR)

**Comment by giswqs** [comment]
@dhirschfeld I have opened an issue on the autocfg repo, suggesting adding the LICENSE file in the next release. Not sure when the maintainer will do it. It is beyond my control. Hope this recipe can be accepted soon so that I can submit the downstream package recipes to get autogluon on conda-forge. Thanks. 

https://github.com/zhreshold/autocfg/issues/2

**Comment by giswqs** [comment]
@conda-forge-admin, please ping team
It recipe has passed all checks and it is ready to be reviewed and merged. Thanks.

File: recipes/autocfg/meta.yaml
Comment (dhirschfeld) [inline recipes/autocfg/meta.yaml:13]:
```suggestion
```
CF don't build for legacy python to this isn't required.
Upstream specify `py36` as the minimum so you should specify that constraint instead:
https://github.com/zhreshold/autocfg/blob/7fb8311bfaea350e8bdcc7010e2a19c9859989d4/setup.py#L21

**Review by dhirschfeld — APPROVED** [review]

File: recipes/autocfg/meta.yaml
Comment (giswqs) [reply recipes/autocfg/meta.yaml]: @dhirschfeld This line has been removed as suggested. Thanks.

File: recipes/autocfg/meta.yaml
Comment (ocefpaf) [inline recipes/autocfg/meta.yaml:23]: This can be made noarch if we drop py37 or if we install `dataclasses` everywhere. I never experimented with the latter but we do that for importlib-metadata, etc.  Do you want to give it a go and see if it works?

Comment (BastianZim) [reply recipes/autocfg/meta.yaml:23]: I think it should work: https://conda-forge.org/docs/maintainer/knowledge_base.html#empty-python-packages

Comment (giswqs) [reply recipes/autocfg/meta.yaml]: I have changed to noarch by dropping `[py<37]`

**Comment by giswqs** [comment]
@ocefpaf After changing to noarch, the win build failed due to `torchvision`. Would that be an issue? The builds were successful before changing to noarch.

**Comment by ocefpaf** [comment]
> @ocefpaf After changing to noarch, the win build failed due to `torchvision`. Would that be an issue? The builds were successful before changing to noarch.

Not a problem. That package is not available o Windows but we prefer a noarch package that fails to install than multiple builds in our CIs.

**Pull request merged**
